### PR TITLE
Move appt instructions to confirmed details page

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
The AppointmentInstructions component is being used on several detail pages in the homepage refresh, but it's resulting in confusing modifications to that component and doesn't appear to be necessary. We can replace it with markup inline on the details pages.

- removed AppointmentInstructions.jsx
- added logic to generate the instructions copy

## Testing done
Local and Unit

## Screenshots
image

## Acceptance criteria
- [ ] No visual changes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by jbalboni** [comment]
@LanceGundersen This looks good, would you mind removing the homepage refresh prop from `src/applications/vaos/appointment-list/components/cards/confirmed/AppointmentInstructions.jsx` (and associated code) and making sure that component isn't used in any of the homepage refresh components?

I made this ticket prior to you making the copy of it that you removed, my main hope was to remove the modifications to the one used on the current homepage so that it stayed specific to the current homepage.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/ConfirmedAppointmentDetailsPage.jsx [hunk 1]
```diff
@@ -55,6 +54,23 @@ function formatHeader(appointment) {
   }
 }
 
+function formatInstructions(instructions) {
+  if (!instructions) {
+    return null;
+  }
+
+  return instructions.split('\n').reduce((obj, str) => {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/ConfirmedAppointmentDetailsPage.jsx:62]: It looks like this splitting on line breaks logic is new, what's the reason for it?

Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/ConfirmedAppointmentDetailsPage.jsx:62]: Yeah I don't know why I did that in the end, I just adjusted it now.

File: src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ describe('VAOS <CommunityCareAppointmentDetailsPage>', () => {
     expect(screen.getByRole('link', { name: /7 0 3. 5 5 5. 1 2 6 4./ })).to.be
       .ok;
     expect(
-      screen.getByRole('heading', { level: 2, name: /Special instructions/ }),
+      screen.getByRole('heading', { level: 4, name: /Special instructions/ }),
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx:98]: I don't think this should have changed. You might need to check the CommunityCareAppointmentDetailsPage and see if it's using the AppointmentInstructions component for the current homepage and update it if so.

Comment (LanceGundersen) [reply src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx:98]: Fixed and adjusted the instructions logic 
image

File: src/applications/vaos/appointment-list/components/CommunityCareAppointmentDetailsPage.jsx [hunk 3]
```diff
@@ -100,10 +109,12 @@ function CommunityCareAppointmentDetailsPage({
       />
 
       <div className="vads-u-margin-top--3 vaos-appts__block-label">
-        <AppointmentInstructions
-          instructions={instructions}
-          isHomepageRefresh
-        />
+        <div className="vads-u-flex--1 vads-u-margin-bottom--2 vaos-u-word-break--break-word">
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/CommunityCareAppointmentDetailsPage.jsx:112]: Based on the screenshot, I think we have some extra info in the header that shouldn't be there. This `instructions` object structure was made up to make some things easier across different appointment types, but I think it's causing confusion in here, where the logic is much simpler than for VA appointments. I think you could do:

```
{!!appointment.comment && (<div className="vads-u-flex--1 vads-u-margin-bottom--2 vaos-u-word-break--break-word">
    <h2 className="vads-u-font-size--base vads-u-font-family--sans vads-u-margin-bottom--0">
        Special instructions
    </h2>
    <div>{appointment.comment}</div>
</div>)}
```

Comment (LanceGundersen) [reply src/applications/vaos/appointment-list/components/CommunityCareAppointmentDetailsPage.jsx]: Done

**Review by jbalboni — APPROVED** [review]

**Pull request merged**